scripts/iam-tools/aws_creds_check.sh

```diff
@@ -0,0 +1,115 @@
+        echo "WARNING WARNING WARNING WARNING WARNING WARNING WARNING"
+    fi
+
+    [ -f "$expfile" ] && read -r line < "$expfile"
+    line=$((line+0))
+    if [ -z "$ZSH_NAME" ]; then
+        #    like 'date +%s' without a fork/exec
+        now="$(printf '%(%s)T' -1)"
+    else
+        # zsh's printf doesn't have time formatters so we're forced to fork/exec date
+        now="$(date +%s)"
+    fi
+    if [ $((now + 60 * 60 * 24 * 7)) -gt "$line" ]; then
+        if [ "$now" -gt "$line" ]; then
+            echo "AWS credentials have expired. Run the following command to refresh them:"
+        else
+            echo "AWS credentials will expire in $(((line-now)/(60*60*24))) day(s)," \
+                "$(((line-now)%(60*60*24)/(60*60))) hour(s). Run the following command to refresh them:"
+        fi
+        echo "aws_api_key_manager --all"
+
+        return 0
+    fi
+    
+    while IFS=: read -r name num; do
+        accounts["$name"]="$num"
+    done < /etc/openshift_tools/aws_accounts.txt
+    # save the count to tell later if we already had any
+    account_count="${#accounts[@]}"
+    if [ -r "$credfile" ]; then
+        while read -r line; do
+            regex='^\[[a-zA-Z0-9_-]*\]$'
+            if [[ "$line" =~ $regex ]]; then
+                # strip off leading and trailing [ and ]
+                line="${line#\[}"
+                line="${line%\]}"
+                if [ -n "${accounts["$line"]+is_set}" ]; then
+                    if [ -n "$ZSH_NAME" ]; then
+                        # zsh syntax for unset requires quotes on associative array keys, which would break bash
+                        unset 'accounts["$line"]'
+                    else
+                        unset accounts["$line"]
+                    fi
+                elif [ "$line" != "default" ]; then
+                    echo "AWS Credentials exist in $credfile for unrecognized (possibly stale?) account '$line'"
+                fi
+            fi
+        done < "$credfile"
+    fi
+    if [ "$account_count" -eq "${#accounts[@]}" ]; then
+        echo "AWS Credentials missing for all accounts. Add them with the following command:"
+        echo "aws_api_key_manager --all"
+    elif [ "${#accounts[@]}" -gt 0 ]; then
+        echo "AWS Credentials missing for one or more accounts. Add them with the following command:"
+        echo -n "aws_api_key_manager"
+        if [ -n "$ZSH_NAME" ]; then
+            # in zsh associative array keys syntax is (@k) instead of !
+            for name in "${(@k)accounts[@]}"; do
+                echo -n " -p '$name'"
+            done
+        else
+            for name in "${!accounts[@]}"; do
+                echo -n " -p '$name'"
+            done
+        fi
+        echo
+    fi
+}
+
+# Only bother to check aws credentials for non-root users with UID<1000 who are members of libra_ops
+if [ $UID -lt 1000 -a $UID -ne 0 ] && id -nG | grep -qw libra_ops; then
```

Comment:
We put non-human users above UID 1000. Some of these are also part of libra_ops, e.g. opsmedic and autokeys. I don't ever want this stuff running for them.